cloud/endagaweb/tasks.py

```diff
@@ -439,3 +439,20 @@ def req_bts_log(self, obj, retry_delay=60*10, max_retries=432):
         raise
     finally:
       obj.save()
+
+
+@app.task(bind=True)
+def zero_out_subscribers_balance(self):
+    """Subscriber balance zero outs when validity expires.
+
+    This runs this as a periodic task managed by celerybeat.
+    """
+    today = django.utils.timezone.now()
+    subscribers = Subscriber.objects.filter(
+        valid_through__lte=today)
+    if not subscribers:
+        return  # Do nothing
+    credit_balance = crdt.PNCounter("default").serialize()
```

Comment:
you don't need to deal with crdt's. You can use the getter/setter `Subscriber` property `balance`